File: banchus-app/src/main/java/pe/nanamochi/banchus/service/RankingService.java
package pe.nanamochi.banchus.service;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import pe.nanamochi.banchus.database.entity.Stat;
import pe.nanamochi.banchus.database.entity.User;
import pe.nanamochi.banchus.domain.enums.CountryCode;
import pe.nanamochi.banchus.domain.enums.Mode;
import pe.nanamochi.banchus.redis.repository.RankingRepository;

@Service
@RequiredArgsConstructor
public class RankingService {
  private final RankingRepository rankingRepository;

  public long getGlobalRank(Mode mode, User user) {
    return rankingRepository.getGlobalRank(mode, user);
  }

  public long getCountryRank(Mode mode, User user, CountryCode countryCode) {
    return rankingRepository.getCountryRank(mode, user, countryCode);
  }

  public void update(Mode mode, User user, Stat stat) {
    rankingRepository.updateRanking(mode, user, stat);
  }
}

File: banchus-app/src/main/kotlin/pe/nanamochi/banchus/service/RankingService.kt
package pe.nanamochi.banchus.service

import org.springframework.stereotype.Service
import pe.nanamochi.banchus.database.entity.Stat
import pe.nanamochi.banchus.database.entity.User
import pe.nanamochi.banchus.domain.enums.CountryCode
import pe.nanamochi.banchus.domain.enums.Mode
import pe.nanamochi.banchus.redis.repository.RankingRepository

@Service
class RankingService(private val rankingRepository: RankingRepository) {
    fun getGlobalRank(mode: Mode, user: User): UInt = rankingRepository.getGlobalRank(mode, user)

    fun getCountryRank(mode: Mode, user: User, countryCode: CountryCode): UInt =
        rankingRepository.getCountryRank(mode, user, countryCode)

    fun update(mode: Mode, user: User, stat: Stat) =
        rankingRepository.updateRanking(mode, user, stat)
}
